src/typeDefs/inputs.ts

```diff
@@ -282,6 +282,15 @@ export const inputs = gql`
     limit: PositiveInt!
   }
 
+  input UpdateAdvertisementInput {
+    id: ID!
+    name: String
+    link: String
+    type: String
+    startDate: String
```

Comment:
Didn't notice this, make use of https://the-guild.dev/graphql/scalars/docs/scalars/date for dates, if you need the time information as well make use of https://the-guild.dev/graphql/scalars/docs/scalars/date-time

Reply:
Add a custom scalar called Start Date and make sure that scalar should be greater than Date.now()
If not then throw appropriate error. Here is the doc for help: https://www.apollographql.com/docs/apollo-server/schema/custom-scalars/